bound/kt/src/test/kotlin/tbdex/sdk/vectors/TbdexTestVectorsProtocolTest.kt

```diff
@@ -14,56 +14,64 @@ class TbdexTestVectorsProtocolTest {
      */
     @Test
     fun parse_order() {
-        testVector("parse-order.json", ::Order) { it.toJson() }
+        testVector("parse-order.json", Order.Companion::fromJsonString) { it.toJsonString() }
     }
 
     @Test
     fun parse_orderstatus() {
-        testVector("parse-orderstatus.json", ::OrderStatus) { it.toJson() }
+        testVector("parse-orderstatus.json", OrderStatus.Companion::fromJsonString) { it.toJsonString() }
     }
 
     @Test
     fun parse_rfq() {
-        testVector("parse-rfq.json", { input -> Rfq(input, true) }) { it.toJson() }
+        testVector("parse-rfq.json", { input ->
+            val rfq = Rfq.fromJsonString(input)
+            rfq.verifyAllPrivateData()
+            rfq
+        }, { it.toJsonString() })
     }
 
     @Test
     fun parse_rfq_omit_private_data() {
-        testVector("parse-rfq-omit-private-data.json", { input -> Rfq(input, false) }) { it.toJson() }
+        testVector("parse-rfq-omit-private-data.json", { input ->
+            val rfq = Rfq.fromJsonString(input)
+            rfq.verifyPresentPrivateData()
+            rfq
+        }, { it.toJsonString() })
     }
 
     @Test
     fun parse_balance() {
-        testVector("parse-balance.json", ::Balance) { it.toJson() }
+        testVector("parse-balance.json", Balance.Companion::fromJsonString) { it.toJsonString() }
     }
 
     @Test
     fun parse_offering() {
-        testVector("parse-offering.json", ::Offering) { it.toJson() }
+        testVector("parse-offering.json", Offering.Companion::fromJsonString) { it.toJsonString() }
     }
 
     @Test
     fun parse_quote() {
-        testVector("parse-quote.json", ::Quote) { it.toJson() }
+        testVector("parse-quote.json", Quote.Companion::fromJsonString) { it.toJsonString() }
     }
 
     @Test
     fun parse_close() {
-        testVector("parse-close.json", ::Close) { it.toJson() }
+        testVector("parse-close.json", Close.Companion::fromJsonString) { it.toJsonString() }
     }
 
     @Test
     fun parse_cancel() {
-        testVector("parse-cancel.json", ::Cancel) { it.toJson() }
+        testVector("parse-cancel.json", Cancel.Companion::fromJsonString) { it.toJsonString() }
     }
 
-    private fun <T> testVector(vectorFileName: String, objectCreation: (String) -> T, toJson: (T) -> String) {
+    private fun <T> testVector(vectorFileName: String, objectCreation: (String) -> T, toJsonString: (T) -> String) {
         val vector = TestVectors.getVector(vectorFileName)
         assertNotNull(vector)
 
         val input = vector!!.get("input").textValue()
         val obj = objectCreation(input)
```

Comment:
don't we want to call verify() here?

I had this in my branch, you pass the objects verify function in

Reply:
ah yeah good point, but one caveat with current implementation is that `verify()` is not on the `Message` nor `Resource` interface. those two interfaces are primarily acting as a type enumeration. but I think we should introduce verification at the interface level, whereafter yes that would work here at this particular spot. until then we'd have to call verify in the calling functions closure, like I did with the RFQ's `verifyAllPrivateData()` and `verifyPresentPrivateData()`